src/goto-synthesizer/enumerative_loop_contracts_synthesizer.cpp

```diff
@@ -75,6 +75,7 @@ std::vector<exprt> construct_terminals(const std::set<symbol_exprt> &symbols)
   }
   result.push_back(from_integer(1, unsigned_int_type()));
   result.push_back(from_integer(1, unsigned_long_int_type()));
+  result.push_back(from_integer(0, signed_int_type()));
```

Comment:
Why exactly this combination of `0` and `1` with exactly those types? It seems this at least needs a comment.

Reply:
I added a new commit that handle the mix of unsigned and signed with different width (rank) in expression enumerator. So we only need 0 and 1 with lowest rank and signed type, that is, ```signed_short_int_type()```.